examples/js/archive.js

```diff
@@ -73,13 +76,65 @@ function renderImages(file, url) {
 
     placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
     placeButton.innerHTML = 'Place on map';
-
+    image.setAttribute('data-original-image', `${url.replace('metadata', 'download')}/${file.name}`);
     image.src = `${url.replace('metadata', 'download')}/${file.name}`;
     imageRow.classList.add('d-flex', 'justify-content-between', 'align-items-center', 'mb-4', 'pe-5');
     imageRow.append(image, placeButton);
+    imageContainer.appendChild(imageRow);
+    imageCount++;
+  });
+};
+
+// this function is called when there are more than a 100 images and thumbnails in data fetched and also if there are not enough thumbnails per image
+// it takes in three arguments thumbs, url and imgs
+// using defualt function parameters and a logical OR, it is reuseable as a graceful failsafe for when there are no thumbnails for all images.
+const getThumbs = (thumbs = [], url, imgs) => {
+  const display = thumbs || imgs;
+
+  display.forEach((file) => {
+    const imageRow = document.createElement('div');
+    const image = new Image(65, 65);
+    const placeButton = document.createElement('a');
+    fetchedFrom = document.createElement('p');
+    const fetchedFromUrl = document.createElement('a');
+    fetchedFromUrl.setAttribute('href', input.value);
+    fetchedFromUrl.setAttribute('target', '_blank');
+    fetchedFromUrl.innerHTML = 'this Internet Archive Collection';
+    fetchedFrom.appendChild(fetchedFromUrl);
+    const fileName = document.createElement('p');
+    fileName.innerHTML = file.name;
+    fileName.classList.add('m-0');
+    fileName.style.fontSize = '12px';
+
+    placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button', 'mt-1');
+    placeButton.innerHTML = 'Place';
+    placeButton.setAttribute('title', 'Place image on map');
 
+    image.setAttribute('data-original', `${url.replace('metadata', 'download')}/${thumbs ? file.original : file.name}`);
+    image.src = `${url.replace('metadata', 'download')}/${file.name}`;
+    imageRow.classList.add('col-4', 'd-flex', 'flex-column', 'p-2', 'align-items-center');
+    imageRow.append(image, placeButton, fileName);
     imageContainer.appendChild(imageRow);
+    imageContainer.setAttribute('class', 'row');
     imageCount++;
+  });
+};
+
+
+// renderImages function is to render the images fetched from IA
```

Comment:
```suggestion
```

Here, I think you've named the function really clearly so the comment doesn't add anything critical! Thank you!

Reply:
Alright, @jywarren Thank you!